packages/cli-platform-android/src/config/findPackageClassName.ts

```diff
@@ -11,22 +11,43 @@ import glob from 'fast-glob';
 import path from 'path';
 import {unixifyPaths} from '@react-native-community/cli-tools';
 
-export function getMainActivityFiles(folder: string) {
-  return glob.sync('**/+(*.java|*.kt)', {cwd: unixifyPaths(folder)});
+export function getMainActivityFiles(
+  folder: string,
+  includePackage: boolean = true,
+) {
+  let patternArray = [];
+
+  if (includePackage) {
+    patternArray.push('*Package.java', '*Package.kt');
+  } else {
+    patternArray.push('*.java', '*.kt');
+  }
+
+  return glob.sync(`**/+(${patternArray.join('|')})`, {
+    cwd: unixifyPaths(folder),
+  });
 }
 
 export default function getPackageClassName(folder: string) {
-  const files = getMainActivityFiles(folder);
+  let files = getMainActivityFiles(folder);
+  let packages = getClassNameMatches(files, folder);
 
-  const packages = files
-    .map((filePath) => fs.readFileSync(path.join(folder, filePath), 'utf8'))
-    .map(matchClassName)
-    .filter((match) => match);
+  if (!packages.length) {
+    files = getMainActivityFiles(folder, false);
```

Comment:
A bit late to the game but I wonder if globbing for `**/*.{java,kt}` once and filtering in-memory instead of hitting the disk twice is faster for bigger projects.

Reply:
@szymonrybczak keen to measure it and sending a followup if that helps with perf? :)

Reply:
Hm, I gave it a shot but the improvement won't be big because after my checks the amount of libraries that needs to fallback are very small.

Now: 
```
❯ hyperfine "npx react-native config --platform android"
Benchmark 1: npx react-native config --platform android
  Time (mean ± σ):      2.070 s ±  0.264 s    [User: 1.049 s, System: 0.607 s]
  Range (min … max):    1.916 s …  2.814 s    10 runs
```
With executing glob once and filtering in-memory:
```
❯ hyperfine "npx react-native config --platform android"
Benchmark 1: npx react-native config --platform android
  Time (mean ± σ):      2.096 s ±  0.190 s    [User: 1.078 s, System: 0.620 s]
  Range (min … max):    1.931 s …  2.572 s    10 runs
```